cmd/selfupdate.go

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of K9s
+
+package cmd
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"runtime"
+
+	"github.com/creativeprojects/go-selfupdate"
+	"github.com/rs/zerolog/log"
+	"github.com/spf13/cobra"
+)
+
+func selfUpdate() *cobra.Command {
+	command := cobra.Command{
+		Use:   "selfupdate",
+		Short: "Update k9s binary",
+		Long:  "Update k9s binary",
+		Run: func(cmd *cobra.Command, args []string) {
+			checkAndUpdateApp()
+		},
+	}
+
+	return &command
+}
+
+func checkAndUpdateApp() error {
+	latest, err := checkVersion()
+	if err != nil {
+		return err
+	}
+
+	if latest.LessOrEqual(version) {
+		log.Info().Msgf("Current version (%s) is the latest", version)
+		return nil
+	}
+
+	return updateApp(latest)
+}
+
+func updateApp(latest *selfupdate.Release) error {
+	exe, err := os.Executable()
```

Comment:
What if this is a link? 
Also what if this process borks and we leave things in no mans land?
Perhaps best to download the requested rev in a tmp location and if we are cool size/checksum move to the final resting place ie real or link?
Also not sure if we get this for free or not but we should validate checksum to make sure we got real mcCoy?

Reply:
That's why I asked to help me to test all scenarios. I'll test links and let you know later.

About checksums - `selfupdate` library should check sums, but again, I'll test it and let you know later.